rest/admin_api.go

```diff
@@ -915,19 +915,44 @@ func (h *handler) handlePutDbConfigImportFilter() error {
 // In non-persistent mode, the endpoint just removes the database from the node.
 func (h *handler) handleDeleteDB() error {
 	h.assertAdminOnly()
+	dbName := h.PathVar("olddb")
+
+	var bucket string
+	var buckets []string
+
+	dbContext, err := h.server.GetDatabase(h.ctx(), dbName)
+
+	if dbContext != nil {
+		bucket = dbContext.Bucket.GetName()
+	} else if h.server.persistentConfig {
+		buckets, err = h.server.BootstrapContext.Connection.GetConfigBuckets()
+		for _, s := range buckets {
+			var config map[string]interface{}
+			_, err = h.server.BootstrapContext.Connection.GetConfig(s, h.server.Config.Bootstrap.ConfigGroupID, &config)
+			if err != nil && err != base.ErrNotFound {
+				return err
+			}
+			if config["name"] == dbName {
+				bucket = s
+				break
+			}
+		}
```

Comment:
We have a function called `ServerContext.fetchDatabase(..., dbName)` which does approximately the same thing, with some optimisations and handling for per-bucket credentials.

It might be worth seeing if we're able to re-use that via `h.server`

Reply:
Do you have any details on why we can't use fetchDatabase here? We will need to re-implement the same logic for per-bucket credentials at the very least, if we're unable to reuse.

Reply:
FetchDatabase reused in commit https://github.com/couchbase/sync_gateway/pull/5894/commits/251ccc647cf1287e5c150048b2eed9e5efb03f75